datumaro/plugins/ndr.py

```diff
@@ -71,7 +104,7 @@ def __init__(self, extractor,
         # parameter validation before main runs
         if working_subset == duplicated_subset:
             raise ValueError("working_subset == duplicated_subset")
-        if algorithm not in [algo_name.name for algo_name in AlgoList]:
+        if algorithm not in [algo.name for algo in Algorithm]:
```

Comment:
I suggest to use the approach we use in other places like this: accept None, string and Enum in c-tor, convert to Enum and use only it internally.

For example: https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/plugins/voc_format/converter.py#L96

Reply:
Can you tell me specific detail of what c-tor is and use internally mean? 
assert algorithm is None or isinstance(algorithm, (str, Algorithm)) 
Is this code correct?
And do I apply change for next 109~112 line?

Reply:
C-tor = class constructor = `__init__` method.
Internally = in the class implementation. Use only Enum values everywhere, where user interaction is not supposed.

1. Looks right. Also I suggest to substitute a default value, if `None` is passed.
2. It would be nice to be consistent.

Reply:
But if I allowed string type, I cannot handle algorithm with string w/ wrong algorithm name.
Do you have any alternatives?